eng/pipelines/templates/jobs/archetype-sdk-integration.yml

```diff
@@ -127,7 +133,17 @@ jobs:
             TestType: "node"
             DependencyVersion: same
             NodeTestVersion: "12.x"
-            TestResultsFiles: "**/test-results.xml"            
+            TestResultsFiles: "**/test-results.xml"   
+
+        # Add matrix entry for canary testing
+        ${{ if eq(parameters.TestCanary, 'true') }}:
```

Comment:
To you expect for these cases you want to add an additional test configuration or do you want to change the configuration? As in you only care about running in public or canary but not both? If you want both then what you have is probably the best option right now. If you only want one or the other then you can make some tweaks to make this a little more scoped.

Reply:
With this new canary entry in the matrix, the idea is to catch the service regressions before they go into the production(since we have seen instances of this happen before and was unnoticed until users complained).
Other than the new entry, the old matrix would continue to run on production environments, it would be more rigorous and is meant for the SDK testing.